File: request-manager/src/main/java/fii/request/manager/domain/WorkflowStep.java
package fii.request.manager.domain;

import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.Data;
import lombok.Getter;
import lombok.Setter;

import java.util.Set;

@Entity
@Data
public class WorkflowStep {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long workflowStepId;

    @ManyToOne
    @JoinColumn(name="workflow_id")
    private Workflow workflow;

    private String stepType;

    private Long stepNumber;

    private String stepName;

    private String stepDescription;


    @OneToOne(cascade = {CascadeType.ALL})
    @JoinColumn(name="document_id")
    private DocumentRequest documentRequest;

    @OneToMany(mappedBy = "workflowStep", cascade = {CascadeType.ALL})
    private Set<FormField> formFields;

    @OneToOne(cascade = {CascadeType.ALL})
    @JoinColumn(name="editable_html_id")
    private EditableHtml editableHtml;

    @OneToOne(cascade = {CascadeType.ALL})
    @JoinColumn(name="email_step_id")
    private EmailStep emailStep;

    @OneToOne(cascade = {CascadeType.ALL})
    @JoinColumn(name="chatgpt_step_id")
    private ChatGptStep chatGptStep;
}

File: workflow-manager-frontend/app/src/main/java/com/example/workflow_manager_frontend/domain/WorkflowStep.kt
package com.example.workflow_manager_frontend.domain

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity
data class WorkflowStep(
    val document: Document? = null,
    val editableHtml: EditableHtml? = null,
    val formFields: MutableList<FormField>? = null,
    val email: Email? = null,
    val chatGptStep: ChatGptStep? = null,
    val stepDescription: String = "",
    val stepName: String = "",
    var stepNumber: Int = 0,
    val stepType: String = "",
    val workflowId: Int = 0,
    @PrimaryKey val workflowStepId: Int = 0
)